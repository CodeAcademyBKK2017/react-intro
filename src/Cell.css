.Cell {
  color: white;
  background-color: rgb(0, 195, 255);
  width: 50px;
  height: 50px;
  text-transform: uppercase;
  
  display: inline-block;
  margin: 1px;
  border-radius: 5px;
  text-align: center;
  line-height: 50px;
}
