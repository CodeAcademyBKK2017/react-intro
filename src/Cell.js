import React, { Component } from 'react';
import './App.css';

class Cell extends Component {

    render() {
        const title = this.props.title.toUpperCase();
        const btnType = title === 'O'? 'btn-o' : 'btn-x';

        return (
            <div className={`btn ${btnType}`}>{title}</div>
        )
    }
}

export default Cell;