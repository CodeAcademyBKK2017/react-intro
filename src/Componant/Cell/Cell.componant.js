import React,{ Component } from 'react';

class Cell extends Component {
  render() {
    return (
      <div style={{
        width:'80px',
        height: '80px',
        margin: '3px',
        border:'0',
        borderRadius:'2px',
        display : 'inline-block',
        textAlign: 'center',
        justifyContent: 'center',
        backgroundColor:this.props.color,
        color:'white',
        boxShadow:' 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19)'
      }}><h1>{this.props.text}</h1>
      </div>
    );
  }
}

export default Cell
