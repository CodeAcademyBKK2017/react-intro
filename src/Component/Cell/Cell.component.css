.Cell{
    width:80px;
    height: 80px;
    margin: 3px;
    border:0;
    border-radius:2px;
    display : inline-block;
    text-align: center;
    justify-content: center;
    background-color:#2196F3;
    color:white;
    box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19)
}

.Cell2{
    width:80px;
    height: 80px;
    overflow: hidden
}