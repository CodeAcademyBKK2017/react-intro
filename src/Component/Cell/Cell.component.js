import React from 'react';
import PropTypes from 'prop-types';
import noop from 'lodash/noop';
import './Cell.component.css';

const Cell = (props) => (
  <div className="Cell" onClick = {props.cellClickHandle(props.text)}>
    <h1>{props.text}</h1>
  </div>
);

Cell.propTypes = {
  text: PropTypes.string.isRequired,
  cellClickHandle: PropTypes.func.isRequired
};

Cell.defaultProps = {
  text: '',
  cellClickHandle: noop
};

export default Cell;
