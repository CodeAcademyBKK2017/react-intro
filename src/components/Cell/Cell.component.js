import React, { Component } from 'react';
import './Cell.style.css';
import PropTypes from 'prop-types';
import noop from 'lodash/noop';

class Cell extends Component {
	render() {
		const {cellClickHandler, text} = this.props;
		
		return (
			<div className="Cell" onClick={cellClickHandler(text)}>
				{this.props.text}
			</div>
		);
	}
}

Cell.propTypes = {
	text: PropTypes.oneOf(['O', 'X']).isRequired,
	cellClickHandler: PropTypes.func.isRequired
};

Cell.defaultProps = {
	text: 'O',
	cellClickHandler: noop
};

export default Cell;
