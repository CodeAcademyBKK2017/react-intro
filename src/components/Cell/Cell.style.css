
.cell {
    display: flex;
    background-color: rgb(3, 117, 223);
    height: 4vh;
    width: 4vw;
    font-size: 4vw;
    text-align: center;
    justify-content:center; 
    color: white;
    border: 2px solid #131010;
    padding: 10px; 
    margin: 8px;
  }
.disable-cell {
    pointer-events: none;
  }