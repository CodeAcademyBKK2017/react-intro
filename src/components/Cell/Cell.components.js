import React, { Component } from 'react';
import './Cell.css';

class Cell extends Component {
  render() {
    return (
        <div className="cell-box" onClick={this.props.cellClickHandler(this.props.name)}>
            { this.props.name }
        </div>
    );
  }
}

export default Cell;
