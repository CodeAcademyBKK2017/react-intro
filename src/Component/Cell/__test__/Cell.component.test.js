import React from 'react';
import Cell from '../Cell.component';
import renderer from 'react-test-renderer';

test('Cell Snapshot Test',() => {
  const tree = renderer.create(<Cell text={'x'}/>);
  expect(tree).toMatchSnapshot();
});
