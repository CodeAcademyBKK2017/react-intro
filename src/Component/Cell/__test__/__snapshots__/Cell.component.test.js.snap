// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cell Snapshot Test 1`] = `
<div
  className="Cell"
  onClick={undefined}
>
  <h1>
    x
  </h1>
</div>
`;
